modules/log6x-multiline-except.adoc

```diff
@@ -0,0 +1,124 @@
+// Module included in the following assemblies:
+//
+// * observability/logging/logging-6.0/log6x-clf.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="log6x-multiline-except_{context}"]
+= Enabling multi-line exception detection
+
+Enables multi-line error detection of container logs.
+
+[WARNING]
+====
+Enabling this feature could have performance implications and may require additional computing resources or alternate logging solutions.
+====
+
+Log parsers often incorrectly identify separate lines of the same exception as separate exceptions. This leads to extra log entries and an incomplete or inaccurate view of the traced information.
+
+.Example java exception
+[,text]
+----
+java.lang.NullPointerException: Cannot invoke "String.toString()" because "<param1>" is null
+    at testjava.Main.handle(Main.java:47)
+    at testjava.Main.printMe(Main.java:19)
+    at testjava.Main.main(Main.java:10)
+----
+
+* To enable logging to detect multi-line exceptions and reassemble them into a single log entry, ensure that the `ClusterLogForwarder` Custom Resource (CR) contains a `detectMultilineErrors` field, with a value of `true`.
+
+.Example ClusterLogForwarder CR
+[source,yaml]
+----
+apiVersion: "observability.openshift.io/v1"
+kind: ClusterLogForwarder
+metadata:
+  name: <log_forwarder_name>
+  namespace: <log_forwarder_namespace>
+spec:
+  filters:
+  - name: <name>
+    type: detectMultilineException
+  pipelines:
+    - inputRefs:
+        - <input-name>
+      name: <pipeline-name>
+      filterRefs:
+        - <filter-name>
+      outputRefs:
+        - <output-name>
+----
+
+== Details
+When log messages appear as a consecutive sequence forming an exception stack trace, they are combined into a single, unified log record. The first log message's content is replaced with the concatenated content of all the message fields in the sequence.
+
+.Supported languages per collector:
+|===
+|Language | Collector
+
+|Java | ✓
+|JS | ✓
+|Ruby | ✓
+|Python | ✓
+|Golang | ✓
+|PHP | ✓
+|Dart | ✓
+|===
+
+== Troubleshooting
+When enabled, the collector configuration will include a new section with type: `detect_exceptions`
+
+.Example vector configuration section
+----
+[transforms.detect_exceptions_app-logs]
+ type = "detect_exceptions"
+ inputs = ["application"]
+ languages = ["All"]
+ group_by = ["kubernetes.namespace_name","kubernetes.pod_name","kubernetes.container_name"]
+ expire_after_ms = 2000
+ multiline_flush_interval_ms = 1000
+----
+
+// OBSDOCS-1104
+== Modifying log level in collector
+
+You can modify the log level in the collector by setting it to `debug`.
+
+.Example ClusterLogForwarder CR
```

Comment:
@jcantrill @anpingli @r2d2rnd Could you please confirm if we can use the following CLF example to modify the log level in collector in our documentation? If not , could you please provide the example that we can use in our production environment/public docs as is. That would be great help.

I will add callouts once we have consensus on this example.

I referred [this](https://gitlab.cee.redhat.com/aosqe/aosqe-tools/-/blob/master/logging/log_doc/clf-6.0-examples/clf-vector-debug.yaml?ref_type=heads) link to add the following example.

Cc @briandooley

Reply:
This could be a snippet IMO where we only highlight the following. It may also be more effective for presentation: 
```
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: collector
  annotations:
    observability.openshift.io/log-level: debug
```